### DISCUSSION TITLE
GitHub Actions override status checks

### DISCUSSION THREAD

**Discussion by gotbahn** [community/community#25313] [Actions]
<p>I use statuses API and create <a href="https://developer.github.com/v3/repos/statuses/" rel="nofollow noopener">statuses</a> to analyze our builds. It was working smoothly in Travis CI. But now after I migrated the same job to Github Actions I’m facing weird issues. Status checks are overriding each other, despite I pass a unique <code>context</code>.</p>
<p>Here is what I have as soon as status check sent from GitHub action job<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/cc69140b3321b2920e7cca47ffa69b26c649624c.png" data-download-href="/uploads/short-url/taix3ygsbIyg05B8oEJk6uaYwq0.png?dl=1" title="Screenshot 2020-06-24 at 19.36.52" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2461970/181090394-ecee2bc6-c142-41a8-a450-094b9be79fd9.png" alt="Screenshot 2020-06-24 at 19.36.52" data-base62-sha1="taix3ygsbIyg05B8oEJk6uaYwq0" width="690" height="320" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/cc69140b3321b2920e7cca47ffa69b26c649624c_2_690x320.png, https://aws1.discourse-cdn.com/github/optimized/2X/c/cc69140b3321b2920e7cca47ffa69b26c649624c_2_1035x480.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/c/cc69140b3321b2920e7cca47ffa69b26c649624c_2_1380x640.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/cc69140b3321b2920e7cca47ffa69b26c649624c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-06-24 at 19.36.52</span><span class="informations">1976×918 130 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>After the jobs are finished, previous checks are gone and fully overridden</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/ccac91afde8ba1dd45c6d11a56af93103d4ea8c3.png" data-download-href="/uploads/short-url/tcD89DZ65kDDxvRGCxRhT1diSLF.png?dl=1" title="Screenshot 2020-06-24 at 19.37.17" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2461970/181090395-0fdc9145-55e4-4a96-9c09-d995437a726a.png" alt="Screenshot 2020-06-24 at 19.37.17" data-base62-sha1="tcD89DZ65kDDxvRGCxRhT1diSLF" width="690" height="320" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/ccac91afde8ba1dd45c6d11a56af93103d4ea8c3_2_690x320.png, https://aws1.discourse-cdn.com/github/optimized/2X/c/ccac91afde8ba1dd45c6d11a56af93103d4ea8c3_2_1035x480.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/c/ccac91afde8ba1dd45c6d11a56af93103d4ea8c3_2_1380x640.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/ccac91afde8ba1dd45c6d11a56af93103d4ea8c3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-06-24 at 19.37.17</span><span class="informations">1976×918 126 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Any ideas why this is happening and how to fix it?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/godban">@godban</a>,</p>
<p>If your status check sent from Github action job, and use pull_request event as a trigger, new workflow started and the status check will be overridden, it’s by designed.</p>
<p>If you use postman with a PAT to create a commit status(or use other workflow to create commit status,eg: push on master branch), the status check will not be overridden with different context.</p>
<p>Thanks.</p>

**Comment by gotbahn** [comment]
<p>Alright, so <code>github.sha</code> in <code>pull_request</code> is not referring to exact commit, so I switched to <code>github.event.after</code> and it works :partying_face:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/6aacda37c64eea33f748602797401e3e746982c7.png" data-download-href="/uploads/short-url/fdGWGaT2dNSRfcdXtKawWFDAs6P.png?dl=1" title="Screenshot 2020-06-25 at 11.32.42" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2461970/181090397-9f8797f1-b9fc-43c6-8445-eaecfb1a64c4.png" alt="Screenshot 2020-06-25 at 11.32.42" data-base62-sha1="fdGWGaT2dNSRfcdXtKawWFDAs6P" width="690" height="245" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/6/6aacda37c64eea33f748602797401e3e746982c7_2_690x245.png, https://aws1.discourse-cdn.com/github/optimized/2X/6/6aacda37c64eea33f748602797401e3e746982c7_2_1035x367.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/6/6aacda37c64eea33f748602797401e3e746982c7_2_1380x490.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/6aacda37c64eea33f748602797401e3e746982c7_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-06-25 at 11.32.42</span><span class="informations">1698×604 86.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> thanks for pointing</p>
